bundles/framework/announcements/view/AnnouncementsModal.jsx

```diff
@@ -0,0 +1,53 @@
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, Checkbox , Modal } from 'antd';
+import { Message } from 'oskari-ui';
+import { Controller, LocaleConsumer } from 'oskari-ui/util';
+import styled from 'styled-components';
+
+//Pop-up functionality for announcements
+
+const StyledCheckbox = styled(Checkbox)`
+    position: absolute !important;
+    left: 0px !important;
+    padding: 4px 15px !important;
+`;
+
+const AnnouncementsModal = ({ title, content, controller, index, visible, checked }) => {
+
+    const handleOk = (index, title, content, checked) => {
+        controller.handleOk(index, title, content, checked);
```

Comment:
A function named handleOk is very generic on the controller. Maybe rename it as something like setAnnouncementAsSeen() or something along those lines? Also I have a bit of a problem with index being used here. Don't the announcements have an id that could be used instead? Why does the handler need title and content as well?

Reply:
Title and content used to create the localStorage item. Using f.ex. id caused problems because of its general ambiguity. I thought using title as the key is unique enough and also gets reset if an announcement's title is changed. This does still however leave the possibility of changing content of an announcement but users not seeing the pop-up, I will look into that. Index is used for splicing the announcement from the modals array. This seemed to be the easiest way to do it but can be changed to something better if suggested.